File: k-kbk/5주차/1874.py
import sys

input = sys.stdin.readline

n = int(input())
stack = []
result = ""
now = 0
isPossible = True

for _ in range(n):
    x = int(input())

    if x >= now:
        while now < x:
            now += 1
            stack.append(now)
            result += "+"

    last = stack[-1]

    if last == x:
        stack.pop()
        result += "-"
    else:
        isPossible = False
        break

if isPossible:
    print("\n".join(result))
else:
    print("NO")

File: keemsebin/5주차/1874.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

rl.on("line", function (line) {
  input.push(line);
}).on("close", function () {
  let T = Number(input[0]);
  let N = input.slice(1).map(Number);

  let stack = [];
  let result = [];
  let index = 1;

  for (let num of N) {
    while (index <= num) {
      stack.push(index++);
      result.push("+");
    }

    if (num !== stack[stack.length - 1]) {
      return console.log("NO");
    }
    result.push("-");
    stack.pop();
  }

  console.log(result.join("\n"));
});
